teps/0098-workflows.md

```diff
@@ -141,8 +145,40 @@ leveraging Workflows for some common functionality but still relying on their ow
 In addition to extensibility mechanisms, we should explore creating a Workflows conformance spec. This will allow existing (and new!) CI/CD projects
 to become Workflows conformant if they don't want to use the default Workflows implementation, giving end users more flexibility in where to run their Workflows.
 
+### Use of Triggers
+
+Some Workflows-like projects built on top of Tekton Pipelines don't make use of Tekton Triggers,
+despite implementing similar functionality. Some Triggers pain points include:
+
+- **Poor performance**: It may take too long to create a PipelineRun after an event is triggered,
+  causing confusion to users who (for example) expect to see a CI process being kicked off immediately
+  after pushing a commit.
+- **Minimal opportunities for customization**: Some projects need the ability to mutate PipelineRuns at
+  runtime or control their creation, instead of just templating PipelineRuns. For example, some projects
+  have found it easier to reimplement Triggers-like functionality than to build concurrency controls on top
+  of Triggers.
+- **Scalability**: A large number of workflows can result in a large number of Triggers, slowing down event
+  processing (due to having to process each of an EventListener's Triggers) and the cluster in general
+  (due to having a large number of CRDs). (TriggerGroups may help to address this need.)
+- **Expressiveness**: It can be more difficult to express functionality via CEL interceptors on Triggers,
+  compared to building features in a conventional programming language.
+  (Custom interceptors may address this need.)
+- **Event payload is not persisted**: A system built on top of Triggers may need access to the event payload
+  (for example, for use in updating an SCM with the status of a CI PipelineRun) without passing this information
+  into the PipelineRun itself via parameter substitution.
+- **Metrics support**: Higher-level systems may need more insight into the PipelineRuns created by the Triggers they use.
+
+Building a successful Workflows project may involve addressing some of these gaps in Triggers, and adding new
+features such as [scheduled and polling runs](./0083-scheduled-and-polling-runs-in-tekton.md).
+
+It's possible that some of these pain points are specific to the way these projects have used Triggers,
+or that newly added features may have addressed some of these pain points.
+More investigation is needed here.
```

Comment:
I'm going to spend a bit more time trying to understand how PAC was previously using triggers and problems you were running into-- linking [https://github.com/openshift-pipelines/pipelines-as-code/tree/release-0.5.0/config](https://www.google.com/url?q=https://github.com/openshift-pipelines/pipelines-as-code/tree/release-0.5.0/config&sa=D&source=docs&ust=1671471923152611&usg=AOvVaw0gaDvH5ygtor-HfVQrHhpE) here for my reference (thanks for this!).

Did PAC ever use a triggers custom interceptor instead of writing trigger bindings for different providers?

Reply:
@lbernick we did evaluate writing a custom interceptor, but if we were going to go down that way and have the interceptor being a catch-all for everything it didn't make sense to keep triggers since we would not use any of the features except for answering to the https request and pass down the payload to the interceptor.